website/profile/utils.py

```diff
@@ -107,6 +111,11 @@ def serialize_contributors(contribs, node, **kwargs):
         for contrib in contribs
     ]
 
+def serialize_visible_contributors(node):
+    # This is optimized when node has .include('contributor__user__guids')
+    return [
+        serialize_user(c, node) for c in node.contributor_set.all() if c.visible
```

Comment:
It shouldn't reduce the query count. Just the data coming back from the server and the number of iterations in that comp. Nothing that's not visible would be transferred or iterated through.